### PR TITLE
*: Use the new CI script as default

### LINKED ISSUE #31893: integration with the CI script

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #31893
A subtask of #30822

Problem Summary:

This is the final step of the "Make CI great again" task: replace the old script with the new one.
Please review other PRs under the #30822 first, most of them are split from this one.

After everything is ready, I'll update this PR 

### What is changed and how it works?

Use the new CI script to replace the old one.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test

Everytime the CI run, this script is executed.

- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle_test.go [hunk 1]
```diff
@@ -950,6 +950,7 @@ func (s *testSerialStatsSuite) checkForGlobalStatsWithOpts(c *C, tk *testkit.Tes
 }
 
 func (s *testSerialStatsSuite) TestAnalyzeGlobalStatsWithOpts1(c *C) {
+	c.Skip("unstable test")
```
Comment (tiancaiamao) [inline statistics/handle/handle_test.go:953]: This test is itself unstable, you can try this many times, and may encounter failure:

```
make ut X='run statistics/handle testSerialStatsSuite.TestAnalyzeGlobalStatsWithOpts1'
```

File: statistics/handle/handle_test.go [hunk 2]
```diff
@@ -989,6 +990,7 @@ func (s *testSerialStatsSuite) TestAnalyzeGlobalStatsWithOpts1(c *C) {
 }
 
 func (s *testSerialStatsSuite) TestAnalyzeGlobalStatsWithOpts2(c *C) {
+	c.Skip("unstable test")
```
Comment (tiancaiamao) [inline statistics/handle/handle_test.go:993]: ditto

File: server/server_test.go [hunk 3]
```diff
@@ -400,6 +400,7 @@ func (cli *testServerClient) runTestLoadDataWithSelectIntoOutfile(t *testing.T,
 }
 
 func (cli *testServerClient) runTestLoadDataForSlowLog(t *testing.T, server *Server) {
+	t.Skip("unstable test")
```
Comment (tiancaiamao) [inline server/server_test.go:403]: This one is weird, it fails in CI env but can't reproduce on my own machine.
I can file a issue for it and add back it later ... but I don't want this single case block the CI replacement...

Comment (tiancaiamao) [reply server/server_test.go:403]: https://github.com/pingcap/tidb/issues/32034

File: statistics/handle/handle_test.go (hunk 1 above)
Comment (tiancaiamao) [reply statistics/handle/handle_test.go:953]: https://github.com/pingcap/tidb/issues/32035

**Review by hawkingrei — APPROVED** [review]

File: tools/check/ut.go [hunk 4]
```diff
@@ -549,7 +553,16 @@ func failureCases(input []JUnitTestCase) int {
...
+func (n *numa) testCommand(pkg string, fn string, old bool) *exec.Cmd {
+	args := make([]string, 0, 10)
+	exe := "./" + testFileName(pkg)
+	if coverprofile != "" {
+		fileName := strings.ReplaceAll(pkg, "/", "_") + "." + fn
+		tmpFile := path.Join(coverFileTempDir, fileName)
+		args = append(args, "-test.coverprofile", tmpFile)
+	}
+	args = append(args, "-test.cpu", "1")
+	// args = append(args, []string{"-test.timeout", "20s"}...)
```
Comment (tiancaiamao) [inline tools/check/ut.go:565]: Just leave the comment here...
I would like to add the limitation in the future.

File: tools/check/ut.go [hunk 5]
```diff
@@ -501,6 +503,8 @@ func (n *numa) runTestCase(pkg string, fn string, old bool) testResult {
 		res.d = time.Since(start)
 		res.Time = formatDurationAsSeconds(res.d)
 	}
+	res.d = time.Since(start)
+	res.Time = formatDurationAsSeconds(res.d)
```
Comment (tangenta) [inline tools/check/ut.go:507]: How about removing line 503-504?

Comment (tiancaiamao) [reply tools/check/ut.go:507]: Done

**Comment by tiancaiamao** [comment]
/run-unit-test

https://github.com/pingcap/tidb/pull/32039

**Review by tangenta — APPROVED** [review]

**Comment by tiancaiamao** [comment]
/run-unit-test

https://github.com/pingcap/tidb/issues/32040

**Pull request merged**